p2p/ipld/read.go

```diff
@@ -3,16 +3,224 @@ package ipld
 import (
 	"context"
 	"errors"
+	"fmt"
 	"math"
+	"math/rand"
 
 	"github.com/ipfs/go-cid"
 	coreiface "github.com/ipfs/interface-go-ipfs-core"
 	"github.com/ipfs/interface-go-ipfs-core/path"
+	"github.com/lazyledger/lazyledger-core/p2p/ipld/plugin/nodes"
+	"github.com/lazyledger/lazyledger-core/types"
+	"github.com/lazyledger/rsmt2d"
 )
 
-// /////////////////////////////////////
+// //////////////////////////////////////
+//	Retrieve Block Data
+// ////////////////////////////////////
```

Comment:
I'm not a big fan of these asci-art separators. if there is a larger chunk of logic that needs to be separated, files or packages are the way we should ideally structure these. IMO, let's change this in this PR but take care of this in a follow-up PR instead.